sip/sip-003-peer-network.md

```diff
@@ -804,22 +804,21 @@ The peer may take an operator-given public IP address.  If no public IP address
 is given, the peer will learn the IP address using the `NatPunchRequest` and
 `NatPunchReply` messages as follows:
 
-1. The peer sends a `NatPunchRequest` to a randomly-chosen neighbor it has
+1. The peer sends a `NatPunchRequest` to a randomly-chosen initial neighbor it has
    already handshaked with.  It uses a random nonce value.
 2. The remote neighbor replies with a (signed) `NatPunchReply` message, with its
    `addrbytes` and `port` set to what it believes the public IP is (based on the
    underlying socket's peer address).
-3. The peer, upon receipt and authentication of the `NatPunchReply`, will
-   confirm the public IP address by attempting to connect to itself.  To do so,
-   it sends a `NatPunchRequest` to the public IP address it learned (with a
-   random nonce).
-4. If the peer successfully connects to itself via the public IP address, it
-   will send a `NatPunchReply` back to itself with the nonce used in step 3.
-5. Upon receipt of the `NatPunchReply`, the peer will have confirmed its public
+3. Upon receipt of the `NatPunchReply`, the peer will have confirmed its public
```

Comment:
In the initial PR https://github.com/blockstack/stacks-blockchain/pull/1642, we touched on the important of step 3 and 4.

<img width="963" alt="Screenshot 2020-07-27 at 4 36 34 PM" src="https://user-images.githubusercontent.com/87777/88589562-59603280-d027-11ea-8321-adea1935e20d.png">

Is this revisited protocol introducing a flaw?

Reply:
This PR makes it so that instead of asking a random neighbor in step 1, the peer only asks one of its initial bootstrapping peers (which it already trusts to tell it other useful network information -- in particular, other neighbors).